eudemo/eudemo/maintenance/lower_port/duct_designer.py

```diff
@@ -246,29 +249,20 @@ def _calc_y_point(x_point):
                 "duct wall thickness would help."
             )
 
-        return make_polygon(
-            [
-                [
-                    ib_div_pt_padded[0],
-                    ib_div_pt_padded[0],
-                    ob_div_pt_padded[0],
-                    ob_div_pt_padded[0],
-                ],
-                [
-                    ib_inner_y,
-                    -ib_inner_y,
-                    -ob_inner_y,
-                    ob_inner_y,
-                ],
-                [
-                    ib_div_pt_padded[1],
-                    ib_div_pt_padded[1],
-                    ob_div_pt_padded[1],
-                    ob_div_pt_padded[1],
-                ],
-            ],
-            closed=True,
-        )
+        ib_div_pt_x, ib_div_pt_z = ib_div_pt_padded
+        ob_div_pt_x, ob_div_pt_z = ob_div_pt_padded
+
+        x = [ib_div_pt_x, ib_div_pt_x, ob_div_pt_x, ob_div_pt_x]
+        y = [ib_inner_y, -ib_inner_y, -ob_inner_y, ob_inner_y]
+        z = [ib_div_pt_z, ib_div_pt_z, ob_div_pt_z, ob_div_pt_z]
+        duct_inner_xy = make_polygon({"x": x, "y": y, "z": z}, closed=True)
+
+        # Translate a little inwards and upwards to ensure penetration to
+        # main body
+        angle = np.deg2rad(self.params.lower_port_angle.value)
+        direction = np.array([np.cos(angle), 0, np.sin(angle)])
+        duct_inner_xy.translate(-1 * direction)
```

Comment:
Not sure why you need to do this as this boundary is constructed around the divertor, which is definatly within the main body.
The angled part of the duct made from this boundary needs to be cut/merged into the blanket, which it defintly goes through (as it's started around the divertor)

Reply:
It just needs to overshoot a bit, otherwise boolean operations from faces which are offset from the original do not fully go through.